src/foundry/foundry.js/pixi/containers/placeableObjects/token.d.ts

```diff
@@ -563,25 +490,37 @@ declare class Token extends PlaceableObject<Token.Data> {
   protected _onDragLeftMove(event: PIXI.InteractionEvent): void;
 
   /**
-   * A factory method to create a Token instance from an Actor entity.
-   * The Token is not automatically saved to the database, it is up to the caller whether or not they wish to do that.
-   *
-   * @param actor     - The input actor entity
-   * @param tokenData - Additional data, such as x, y, rotation, etc. for the created token
-   * @returns The created Token instance
+   * @deprecated since 0.8.0
    */
-  static fromActor(actor: Actor, tokenData?: DeepPartial<Token.Data>): Promise<Token>;
+  static fromActor(actor: Actor, tokenData?: DeepPartial<Token.Data>): never;
 
   /**
-   * @deprecated since 0.7.4
+   * A helper method to retrieve the underlying data behind one of the Token's attribute bars
+   * @param barName     - The named bar to retrieve the attribute for
+   * @param alternative - An alternative attribute path to get instead of the default one
+   * @returns The attribute displayed on the Token bar, if any
+   * @deprecated since 0.8.0
    */
-  shiftPosition(dx: number, dy: number): Promise<this>;
+  getBarAttribute(
+    barName: string,
+    { alternative }?: { alternative?: string }
+  ): {
+    type: 'bar' | 'value';
+    attribute: string;
+    value: number;
+    max?: number;
+  } | null;
+
+  /** @override */
+  clone(): Token;
+
+  /** @override */
+  get bounds(): NormalizedRectangle;
 
   /**
-   * @deprecated since 0.7.4
-   * @see {@link Token#toggleEffect}
+   * @remarks Not used for `Token`
    */
-  toggleOverlay(texture: string | ActiveEffect.Data): Promise<boolean>;
+  controlIcon: null;
 }
 
 declare namespace Token {
```

Comment:
Should the `Data`-Interface stay, or must it be moved to another Location like `TokenData`?
Also: How should the `actorData` be typed?

Reply:
Am I right that it would make sense to add the `BaseToken` and its scheme to this PR et move the data there?

Reply:
The `Data` interface in here will likely get replaced with the `TokenData` document schema and friends. I already started working on the `PrototypeTokenData`, which also caused me to work on `TokenData` and `TokenBarData`. So let me finish that up and you can use it here. But feel free to add `BaseToken`, if it's already feasible to add it.